diff --git a/src/plugins/notification/notification_core.c b/src/plugins/notification/notification_core.c
index 6831068..73076d7 100644
--- a/src/plugins/notification/notification_core.c
+++ b/src/plugins/notification/notification_core.c
@@ -286,7 +286,8 @@ gboolean notification_notified_hash_msginfo_update(MsgInfoUpdate *msg_update)
   g_return_val_if_fail(msg_update != NULL, FALSE);
 
   if((msg_update->flags & MSGINFO_UPDATE_FLAGS) &&
-     !MSG_IS_NEW(msg_update->msginfo->flags)) {
+     !MSG_IS_NEW(msg_update->msginfo->flags) &&
+     !MSG_IS_UNREAD(msg_update->msginfo->flags)) {
 
     MsgInfo *msg;
     gchar *msgid;
@@ -340,15 +341,15 @@ static gboolean notification_traverse_hash_startup(GNode *node, gpointer data)
   FolderItem *item = (FolderItem*) node->data;
   gint new_msgs_left;
 
-  if(!(item->new_msgs))
+  if(!(item->new_msgs + item->unread_msgs))
     return FALSE;
 
-  new_msgs_left = item->new_msgs;
+  new_msgs_left = item->new_msgs + item->unread_msgs;
   msg_list = folder_item_get_msg_list(item);
 
   for(walk = msg_list; walk; walk = g_slist_next(walk)) {
     MsgInfo *msg = (MsgInfo*) walk->data;
-    if(MSG_IS_NEW(msg->flags)) {
+    if(MSG_IS_NEW(msg->flags) || MSG_IS_UNREAD(msg->flags)) {
       gchar *msgid;
 
       if(msg->msgid)
@@ -398,7 +399,7 @@ void notification_new_unnotified_msgs(FolderItemUpdateData *update_data)
     MsgInfo *msg;
     msg = (MsgInfo*) walk->data;
 
-    if(MSG_IS_NEW(msg->flags)) {
+    if(MSG_IS_NEW(msg->flags) || MSG_IS_UNREAD(msg->flags)) {
       gchar *msgid;
 
       if(msg->msgid)
diff --git a/src/plugins/notification/notification_popup.c b/src/plugins/notification/notification_popup.c
index c69c8c7..e66ece5 100644
--- a/src/plugins/notification/notification_popup.c
+++ b/src/plugins/notification/notification_popup.c
@@ -101,6 +101,9 @@ void notification_popup_msg(MsgInfo *msginfo)
   if(!msginfo || !notify_config.popup_show)
     return;
 
+  if(!notify_config.popup_unread_also && !MSG_IS_NEW(msginfo->flags))
+    return;
+
   if(notify_config.popup_folder_specific) {
     guint id;
     GSList *list;
diff --git a/src/plugins/notification/notification_prefs.c b/src/plugins/notification/notification_prefs.c
index 52e3a3a..49f2305 100644
--- a/src/plugins/notification/notification_prefs.c
+++ b/src/plugins/notification/notification_prefs.c
@@ -105,6 +105,7 @@ typedef struct {
 	GtkWidget *popup_cont_color_sel;
 #else /* HAVE_LIBNOTIFY */
 	GtkWidget *popup_display_folder_name;
+	GtkWidget *popup_unread_also;
 #endif /* HAVE_LIBNOTIFY */
 }NotifyPopupPage;
 NotifyPopupPage popup_page;
@@ -246,6 +247,9 @@ PrefParam
 #else /* HAVE_LIBNOTIFY */
 				{	"popup_display_folder_name", "FALSE", &notify_config.popup_display_folder_name,
 					P_BOOL, NULL, NULL, NULL},
+        { "popup_unread_also", "FALSE", &notify_config.popup_unread_also,
+          P_BOOL, NULL, NULL, NULL},
+
 #endif /* HAVE_LIBNOTIFY */
 #endif
 
@@ -1120,6 +1124,17 @@ static void notify_create_popup_page(PrefsPage *page, GtkWindow *window,
 	gtk_widget_show(checkbox);
 	gtk_widget_show(hbox);
 	popup_page.popup_display_folder_name = checkbox;
+
+  hbox = gtk_hbox_new(FALSE, 10);
+  gtk_box_pack_start(GTK_BOX(vbox), hbox, FALSE, FALSE, 0);
+  checkbox = gtk_check_button_new_with_label(_("Notify on unread also"));
+  gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(checkbox),
+      notify_config.popup_unread_also);
+  gtk_box_pack_start(GTK_BOX(hbox), checkbox, FALSE, FALSE, 0);
+  gtk_widget_show(checkbox);
+  gtk_widget_show(hbox);
+  popup_page.popup_unread_also = checkbox;
+
 #endif /* HAVE_LIBNOTIFY */
 
 	notify_popup_enable_set_sensitivity
@@ -1168,6 +1183,9 @@ static void notify_save_popup(PrefsPage *page)
 #else /* HAVE_LIBNOTIFY */
 	notify_config.popup_display_folder_name =
 	gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(popup_page.popup_display_folder_name));
+  notify_config.popup_unread_also =
+  gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(popup_page.popup_unread_also));
+
 #endif /* HAVE_LIBNOTIFY */
 }
 
diff --git a/src/plugins/notification/notification_prefs.h b/src/plugins/notification/notification_prefs.h
index 684f6ca..8a5d626 100644
--- a/src/plugins/notification/notification_prefs.h
+++ b/src/plugins/notification/notification_prefs.h
@@ -72,6 +72,7 @@ typedef struct {
   gulong           popup_color_fg;
 #else /* HAVE_LIBNOTIFY */
   gboolean         popup_display_folder_name;
+  gboolean         popup_unread_also;
 #endif /* HAVE_LIBNOTIFY */
 #endif /* Popup */
 #ifdef NOTIFICATION_COMMAND
